src/OrchardCore/OrchardCore.Localization.Core/DefaultPluralRuleProvider.cs

```diff
@@ -52,17 +52,30 @@ bool IPluralRuleProvider.TryGetRule(CultureInfo culture, out PluralizationRuleDe
         {
             rule = null;
 
-            if (Rules.TryGetValue(culture.Name, out rule))
-            {
-                return true;
-            }
+            var cultureForPlural = GetBaseCulture(culture);
 
-            if (culture.Parent != null && Rules.TryGetValue(culture.Parent.Name, out rule))
+            if (Rules.TryGetValue(cultureForPlural.Name, out rule))
             {
                 return true;
             }
 
             return false;
         }
+
+
+        /// <summary>
+        /// Get the base culture of the provided culture.
+        /// e.g., zh-Hans-CN -> zh-Hans -> zh
+        /// </summary>
+        private CultureInfo GetBaseCulture(CultureInfo culture)
```

Comment:
Can we take first two letters instead of loop? Because the intend is to get the base culture name

Reply:
I am not sure. I am not familiar with culture stuff.  And I see 3 letters keys in the matrix.